auslib/blobs/apprelease.py

```diff
@@ -10,6 +11,22 @@ class ReleaseBlobBase(Blob):
     def __init__(self, **kwargs):
         Blob.__init__(self, **kwargs)
 
+    def validate(self, product, whitelistedDomains):
+        Blob.validate(self, product, whitelistedDomains)
+        requiredLen = getHashLen(self["hashFunction"])
+        for platform in self.get("platforms", {}).values():
+            for locale in platform.get("locales", {}).values():
+                for partial in locale.get("partials", {}):
+                    actualLen = len(partial["hashValue"])
+                    if actualLen != requiredLen:
+                        raise ValueError("The hashValue length is different from the required length of {} for {}."
+                                         .format(getHashLen(self["hashFunction"]), self["hashFunction"].lower()))
+                for complete in locale.get("completes", {}):
+                    actualLen = len(complete["hashValue"])
+                    if actualLen != requiredLen:
+                        raise ValueError("The hashValue length is different from the required length of {} for {}."
+                                         .format(getHashLen(self["hashFunction"]), self["hashFunction"].lower()))
+
```

Comment:
This logic will need to move elsewhere - the format being assumed here is not shared across all children of ReleaseBlobBase, so it's not appropriate to put it here. The two formats here are implemented by SingleUpdateXMLMixin and MultipleUpdatesXMLMixin so my suggestion is to put this logic in a helper method (eg, _validateHashLengths) in each of those classes, and call that method from ReleaseBlobBase.validate().